Tools/dea_tools/coastal.py

```diff
@@ -407,6 +412,172 @@ def _model_tides(
     return tide_df
 
 
+def _ensemble_model(
+    x,
+    y,
+    crs,
+    tide_df,
+    ensemble_models,
+    ensemble_func=None,
+    ensemble_top_n=3,
+    ranking_points="https://dea-public-data-dev.s3-ap-southeast-2.amazonaws.com/derivative/dea_intertidal/supplementary/rankings_ensemble_2017-2019.geojson",
```

Comment:
These points are labelled 2017-2019. Are they the altimetry/tide gauge elevations? Any thoughts/plans for having multitemporal functionality here?

Reply:
Yeah, that file includes the rankings of each tide model model at both the satellite altimetry points, and the NDWI-tide correlation points. Both those comparisons were done over the same 2017-2019 for a few reasons:

- Maximise signal-to-noise by selecting a relatively calm period for SOI-driven sea level dynamics (El Nino/La Nina) 
- Overlap with Sentinel-2 (particularly the post-2017 Sentinel-2A + 2B era) 
- Incorporate altimetry data from missions that were not used to develop any of the any of the tide models being evaluated 

So the "hope" is that this period is pretty representative, and rankings from it can be reasonably applied to other periods too. But you could easily calculate the rankings for any period you wanted and pass them in here instead!

Reply:
These are the two notebooks that generate the rankings:

- https://github.com/GeoscienceAustralia/dea-intertidal/blob/rbt/notebooks/ensemble_tides/Tide_correlation_rankings.ipynb
- https://github.com/GeoscienceAustralia/dea-intertidal/blob/rbt/notebooks/ensemble_tides/Coastal_altimetry_rankings.ipynb

And the notebook that evaluates the ensemble against tide gauge data:

- https://github.com/GeoscienceAustralia/dea-intertidal/blob/rbt/notebooks/ensemble_tides/Tide_gauge_comparison.ipynb